### PR TITLE
Wearable backpacks + HMCCosmetic integration

### PR DESCRIPTION
**Pull request by Boy0000**

### CODE REVIEW

File: mineinabyss-components/src/main/kotlin/com/mineinabyss/components/cosmetics/Cosmetics.kt [hunk 1]
```diff
@@ -10,7 +10,8 @@ import org.bukkit.entity.Player
 @Serializable
 @SerialName("mineinabyss:cosmetics")
 class Cosmetics(
-    var gesture: @Contextual Gesture? = null
+    var gesture: @Contextual Gesture? = null,
```
Comment (0ffz) [inline mineinabyss-components/src/main/kotlin/com/mineinabyss/components/cosmetics/Cosmetics.kt:13]: Best if this isn't a var, same for all other components and mutable lists if they don't need to be changed often, see https://wiki.mineinabyss.com/geary/guide/designing-components/#aim-for-immutable-components

File: mineinabyss-features/src/main/kotlin/com/mineinabyss/cosmetics/CosmeticListener.kt [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+import org.bukkit.inventory.ItemStack
+import org.bukkit.inventory.meta.BlockStateMeta
+
+class CosmeticListener : Listener {
+
+    // Cancel HMCCosmetics backpack equip if player isn't wearing a backpack
+    @EventHandler
+    fun CosmeticChangeEvent.onEquipBackpack() {
+        val player = (user as? User ?: return).player
+        if (cosmeticItem.type == ArmorItem.Type.BACKPACK && !player!!.toGeary().has<Backpack>()) {
```
Comment (0ffz) [inline mineinabyss-features/src/main/kotlin/com/mineinabyss/cosmetics/CosmeticListener.kt:28]: Handle player null case above

File: mineinabyss-features/src/main/kotlin/com/mineinabyss/cosmetics/CosmeticListener.kt [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+            isCancelled = true
+        }
+    }
+
+    @EventHandler
+    fun PlayerInteractEvent.equipBackpack() {
+        if (player.isSneaking && rightClicked) {
+            if (player.getCosmeticBackpack().itemStack.type == Material.AIR) {
+                val i = item?.type.toString().lowercase()
+                val type = if (!i.startsWith("shulker")) i.replace("_shulker_box", "") else i
```
Comment (0ffz) [inline mineinabyss-features/src/main/kotlin/com/mineinabyss/cosmetics/CosmeticListener.kt:39]: A couple comments for this event listener would be nice

**Review by 0ffz — CHANGES_REQUESTED** [review]

File: mineinabyss-components/src/main/kotlin/com/mineinabyss/components/cosmetics/Cosmetics.kt (hunk 1 above)
Comment (Boy0000) [reply mineinabyss-components/src/main/kotlin/com/mineinabyss/components/cosmetics/Cosmetics.kt:13]: Its unused for now so ill remove it

File: mineinabyss-components/src/main/kotlin/com/mineinabyss/components/cosmetics/Cosmetics.kt [hunk 4]
```diff
@@ -1,16 +1,14 @@
...
-import kotlinx.serialization.Contextual
 import kotlinx.serialization.SerialName
 import kotlinx.serialization.Serializable
 import org.bukkit.entity.Player
 
 @Serializable
 @SerialName("mineinabyss:cosmetics")
 class Cosmetics(
-    var gesture: @Contextual Gesture? = null
+    var cosmeticBackpack: String? = null
```
Comment (0ffz) [inline mineinabyss-components/src/main/kotlin/com/mineinabyss/components/cosmetics/Cosmetics.kt:11]: bro this is still var tho

wack

**Review by 0ffz — CHANGES_REQUESTED** [review]

**Comment by Boy0000** [comment]
Readded gesture since i now changed it to fit the wiki link.
Tested and made sure the backpack content is stored across restart

File: mineinabyss-components/src/main/kotlin/com/mineinabyss/components/players/Backpack.kt [hunk 5]
```diff
@@ -0,0 +1,11 @@
+package com.mineinabyss.components.players
+
+import com.mineinabyss.idofront.serialization.SerializableItemStack
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+@Serializable
+@SerialName("mineinabyss:backpack")
+class Backpack(
+    var backpackContent: MutableList<SerializableItemStack>? = mutableListOf()
```
Comment (0ffz) [inline mineinabyss-components/src/main/kotlin/com/mineinabyss/components/players/Backpack.kt:10]: Check all components for what I said. This looks like it'd be fine as a mutable list but var makes no sense

File: mineinabyss-features/src/main/kotlin/com/mineinabyss/helpers/CosmeticHelpers.kt [hunk 6]
```diff
@@ -1,41 +1,33 @@
...
 import io.lumine.cosmetics.players.Profile
-import org.bukkit.Material
 import org.bukkit.entity.Player
-import org.bukkit.inventory.ItemStack
 
 fun Player.playGesture(gesture: String) {
-    mcCosmetics.profiles.getProfile(name).ifPresent { profile: Profile? ->
-        mcCosmetics.gestureManager.getCosmetic(gesture).ifPresent { gesture ->
-            cosmetics.gesture = gesture
+    mcCosmetics.profiles.getProfile(name).ifPresent profile@{ profile: Profile? ->
```
Comment (0ffz) [inline mineinabyss-features/src/main/kotlin/com/mineinabyss/helpers/CosmeticHelpers.kt:14]: You can probably just get or else return, perhaps get or null ?: return.

I generally dislike the extra indentation ifPresent adds

**Review by 0ffz — APPROVED** [review]

**Pull request merged**